src/lib.rs

```diff
@@ -3853,6 +3922,58 @@ mod tests {
         let mut r = pipe.server.readable();
         assert_eq!(r.next(), None);
     }
+
+    #[test]
+    fn stream_round_robin() {
```

Comment:
I think the goal of this test is to check writing order is preserved - write to stream 8, 0, 4 and client read stream 8, 0, 4. Correct? If so it's good to have some comment what this test try to validate.

Reply:
Fair enough. I added descriptions to both new tests, let me know if they need clarification.